src/annotator/adder.js

```diff
@@ -83,24 +83,23 @@ export class Adder {
    *
    * The adder is initially hidden.
    *
-   * @param {HTMLElement} container - The DOM element into which the adder will be created
+   * @param {HTMLElement} element - The DOM element into which the adder will be created
    * @param {AdderOptions} options - Options object specifying `onAnnotate` and `onHighlight`
    *        event handlers.
    */
-  constructor(container, options) {
-    this._container = container;
-    this._shadowRoot = createShadowRoot(container);
+  constructor(element, options) {
+    this._outerContainer = document.createElement('hypothesis-adder');
+    element.appendChild(this._outerContainer);
+    this._shadowRoot = createShadowRoot(this._outerContainer);
```

Comment:
There is a pattern that could potentially be extracted in future for setting up an element with a custom `hypothesis-` name, creating a shadow DOM, rendering a Preact component into it and then cleaning up later.

API sketch:

```js
// In util/preact-container.js
class PreactContainer {
  constructor(element, name) { /* create element and shadow root */ }
  render(vnode) { /* render vnode into shadow root */ }
  remove() { /* unmount and remove container */ }
}

// Usage example
this._container = new PreactContainer(element, 'hypothesis-adder');
this._container.render(<Adder .../>);
this._container.remove();
```

Reply:
Good idea. I like that.